lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -0,0 +1,949 @@
+#include "TypeChecker.h"
+#include "swift/AST/Decl.h"
+#include "swift/AST/Expr.h"
+#include "swift/AST/Module.h"
+#include "swift/AST/ParameterList.h"
+#include "swift/AST/Pattern.h"
+#include "swift/AST/Stmt.h"
+#include "swift/AST/Types.h"
+#include "DerivedConformances.h"
+
+using namespace swift;
+using namespace DerivedConformance;
+
+// Returns whether the type represented by the given ClassDecl inherits from a
+// type which is Codable.
+static bool classInheritsCodable(TypeChecker &tc, ClassDecl *type) {
+  if (!type || !type->hasSuperclass())
```

Comment:
That's handled by default below in `getOrSynthesizeCodingKeysDecl`, `deriveCodable_encode`, and `deriveCodable_init`. This is used to determine whether a class has a `Codable` superclass so we can emit a `super` key, and call `try super.encode(to: container.superEncoder())` and `try super.init(from: container.superDecoder())`; if the class is `Codable` but its superclass is not, we don't need to emit those things.

Perhaps a better name for this function would just be `superclassIsCodable` or similar?